en/guides/typesafehttpbasic.md

```diff
@@ -0,0 +1,172 @@
+<div class="titleBlock">
+  <h1>Type-Safe HTTP basic authentication</h1>
+  <p>How to add HTTP basic authentication support to Kitura's Codable Routing</p>
+</div>
+
+## Introduction
+Kitura 2.4 introduces "Type-safe Middleware": middlewares that have a structure and data types that you define, that are instantiated and passed to your handler upon a request. We also introduce `TypeSafeCredentials`, a protocol that implements `TypeSafeMiddleware` to authenticate a request on a Codable route and initialize itself with the authenticated users data.
+
+HTTP Basic authentication transmits credentials in an "Authorization" header as base64 encoded user ID/password pairs. Many clients also let you send the username and the password in the URL as follows:
+
+```
+https://username:password@www.example.com/
+```
+
+However some web browsers disable this for security reasons.
+
+This guide steps you through using `TypeSafeHTTPBasic`, a `TypeSafeCredentials` plugin for safe and easy HTTP basic authentication in Codable routes.
+
+## Adding TypeSafeHTTPBasic to your project
+
+1) Add `Kitura-CredentialsHTTP` to your `Package.swift` dependencies:
+```swift
+.package(url: "https://github.com/IBM-Swift/Kitura-CredentialsHTTP.git", from: "2.1.0"),
+```
+
+2) Add `CredentialsHTTP` to your Application targets:
+```swift
+.target(name: "Application", dependencies: [ "CredentialsHTTP", ...
+```
+
+3) Regenerate your Xcode project:
+```
+swift package generate-xcodeproj
+```
+
+4) Open your Xcode project and go to Sources >> Application >> Application.swift
+
+5) Add `CredentialsHTTP` to your import statements:
+```swift
+import CredentialsHTTP
+```
+
+## Defining a TypeSafeHTTPBasic object
+
+We will declare a struct which conforms to `TypeSafeHTTPBasic`. This will be initialized when our route is successfully authenticated and we will be able to access the authenticated user's id within our Codable route.
+
+1) Define the TypeSafeHTTPBasic Struct:
+
+At the bottom of `Application.swift`, Define a public struct called `MyBasicAuth` and set it to conforms to `TypeSafeHTTPBasic`:
```

Comment:
Define -> define
set it to conforms to `TypeSafeHTTPBasic` -> that conforms to the `TypeSafeHTTPBasic` protocol: